### ISSUE TITLE
Child collections losing every second child after root SaveAndMergeAsync()

### ISSUE THREAD

**Issue by michaelcsikos** [MarimerLLC/csla#3431]
I upgraded to CSLA 7.0.0 today and I thought I was taking crazy pills. I've set up an example based on the BlazorServerExample:

[BlazorServerExample.zip](https://github.com/MarimerLLC/csla/files/12356829/BlazorServerExample.zip)

**Log in** then click **Add person**. You'll see a list of 10 **Child items** showing a `Guid`.

1. Enter a **Name** and click **Save**.
2. Every second child item has disappeared.
3. Edit the name and click `Save` and the list is halved again.

![image](https://github.com/MarimerLLC/csla/assets/21806535/052a3dbd-b12c-4024-b5cb-d7be614420f8)
![image](https://github.com/MarimerLLC/csla/assets/21806535/057268d1-f88f-4ba4-8a9f-62dd98098aeb)
![image](https://github.com/MarimerLLC/csla/assets/21806535/71121ae5-5527-459f-8c49-ef36e20156f0)
![image](https://github.com/MarimerLLC/csla/assets/21806535/347eeb93-5ed2-4d6a-b946-c99ce90171f7)
![image](https://github.com/MarimerLLC/csla/assets/21806535/2e900d60-b6ca-4247-8173-b69bfd434933)

**Comment by michaelcsikos** [comment]
There's something wrong with the `GraphMerger`.

```cs
    private async Task SaveAsync()
    {
        var model = vm.Model;

        model     = await model.SaveAsync(); // this is fine

        // await model.SaveAndMergeAsync(); this is broken

        vm.Model  = model;

        await InvokeAsync(StateHasChanged);
    }
```

**Comment by michaelcsikos** [comment]
If I roll the NuGet packages in each csproj back to CSLA 6.2.2 it works as expected:

[BlazorServerExample6.2.2.zip](https://github.com/MarimerLLC/csla/files/12357138/BlazorServerExample6.2.2.zip)

**Comment by swegele** [comment]
Honestly I've never used the `SaveAndMergeAsync` method yet.  When reading this issue I thought "That would be really nice because then I don't have to manually handle swapping the new reference returned from the Save method we've used for years".  

Do we have confirmation that this method is broken/not use-able for now?

**Comment by michaelcsikos** [comment]
Hi @swegele if nothing has changed, I would expect the 7.0 and 6.2.2 examples that I posted would exhibit the problem. I have continued to use 6.2.2. I'm not in the habit of using `SaveAndMergeAsync` either, but it's automatically used by the `ViewModel` for Blazor.

**Comment by rockfordlhotka** [comment]
This is resolved by #3481 and is in v7.0.1
